integration_tests/tests/legacy.rs

```diff
@@ -326,3 +328,134 @@ async fn test_kafka_raw() {
+            "?from=2024-12-31T23:00:00Z&to=2025-01-01T01:30:00Z",
+            9999,
+            211,
+            3,
+        ),*/
+    ];
+
+    e2e_test_wrapper_legacy(async |producer: FutureProducer, db_pools: DbPools| {
+        let t1: DateTime<Utc> = Utc.with_ymd_and_hms(2024, 12, 31, 20, 0, 0).unwrap();
+        let test_data = [
+            TestData {
+                station_id: 10001,
+                params: vec![Param::new("TA")],
+                start_time: t1,
+                period: Duration::hours(1),
+                type_id: 508,
+                len: 8,
+            },
+            TestData {
+                station_id: 10001,
+                params: vec![Param::new("TA")],
+                start_time: t1,
+                period: Duration::hours(1),
+                type_id: 501,
+                len: 8,
+            },
+            TestData {
+                station_id: 20001,
+                params: vec![Param::new("TGX")],
+                start_time: t1,
+                period: Duration::hours(1),
+                type_id: 501,
+                len: 8,
+            },
+            TestData {
+                station_id: 9999,
+                params: vec![Param::new("TA")],
+                start_time: t1,
+                period: Duration::hours(1),
+                type_id: 508,
+                len: 8,
+            },
+            TestData {
+                station_id: 9999,
+                params: vec![Param::new("TA")],
+                start_time: t1,
+                period: Duration::hours(1),
+                type_id: 501,
+                len: 8,
+            },
+        ];
+
+        for ts in test_data {
+            producer
+                .send_result(
+                    FutureRecord::to(KAFKA_RAW_TOPIC)
+                        .key("")
+                        .payload(&ts.obsinn_message()),
+                )
+                .unwrap()
+                .await
+                .unwrap()
+                .unwrap();
+        }
+
+        // TODO: we do not have an API endpoint to query the flags.kvdata table
+        let open_conn = db_pools.open.get().await.unwrap();
+
+        // As we have no way to sync with message processing in kvkafka ingestion, we just keep
+        // trying to fetch data with a timeout
+        let timeout = std::time::Duration::from_secs(10);
+        let timeout_start = Instant::now();
+        loop {
+            if let Ok(data_rows) = open_conn
+                .query(
+                    r#"
+                        SELECT
+                            timeseries,
+                            obstime,
+                            original
+                        FROM legacy.data
+                    "#,
+                    &[],
+                )
+                .await
+            {
+                if !data_rows.is_empty() {
+                    break;
+                }
+                // or else keep looping since no data has been written
+            }
```

Comment:
I think there's a race condition here that will make this test flaky. You're only waiting for "any" data to appear, so this can progress after the first kafka message has been ingested. I think you need to add a WHERE condition here to specifically check for the data in the last message you send

Reply:
If you think the last commit is appropriate, then I think it resolves this.